test/e2e/framework/nodes.go

```diff
@@ -28,30 +28,31 @@ import (
 	"k8s.io/apimachinery/pkg/types"
 )
 
-// FindNodesByGatewayLabel finds the nodes in a given cluster by matching 'submariner.io/gateway' value.
-// Nodes with the missing label will be ignored. Note the control plane node labeled as master is ignored.
-func (f *Framework) FindNodesByGatewayLabel(cluster ClusterIndex, isGateway bool) []*v1.Node {
-	return findNodesByGatewayLabel(int(cluster), isGateway)
+// FindGatewayNodes finds nodes in a given cluster by matching 'submariner.io/gateway' value.
+func (f *Framework) FindGatewayNodes(cluster ClusterIndex) []*v1.Node {
+	return findNodesByLabel(int(cluster), GatewayLabel, true)
 }
 
-func findNodesByGatewayLabel(cluster int, isGateway bool) []*v1.Node {
+// FindGatewayNodes finds nodes in a given cluster that doesn't match 'submariner.io/gateway' value.
+func (f *Framework) FindNonGatewayNodes(cluster ClusterIndex) []*v1.Node {
+	return findNodesByLabel(int(cluster), GatewayLabel, false)
```

Comment:
Similarly, using a `LabelSelector`:

```suggestion
	return KubeClients[cluster].CoreV1().Nodes().List(context.TODO(), 
	    metav1.ListOptions{
	               LabelSelector: labels.NewSelector().Add(
				newRequirement(GatewayLabel, selection.Exists, []string{}),
				newRequirement(GatewayLabel, selection.NotEquals, []string{"true"})).String(),
	           })
```

New helper function:

```Go
func newRequirement(key string, op selection.Operator, vals []string) labels.Requirement {
	r, err := labels.NewRequirement(key, op, vals)
	Expect(err).To(Succeed())

	return *r
}
```

Reply:
Suggested changes implemented.
Helper function added.